cmd/kubeadm/app/util/config/common.go

```diff
@@ -190,8 +191,37 @@ func ChooseAPIServerBindAddress(bindAddress net.IP) (net.IP, error) {
 	return ip, nil
 }
 
+// verifyKnownGVKs takes a list of GVKs and verifies if they are known in kubeadm or component config schemes
+func verifyKnownGVKs(gvks []schema.GroupVersionKind) error {
```

Comment:
`documentMapTo{Init|Join}Configuration()`  calls `VerifyUnmarshalStrict()` which contains similar logic to test for known GVKs. having `verifyKnownGVKs()`  and calling it early in `MigrateOldConfig()` and `ValidateConfig()`  has the benefits of existing early for unknown GVKs. logic is a bit duplicated, yet for both migrate/validate commands we want a common method to check if an API is known early on and include component config in there as well. during `VerifyUnmarshalStrict()`  for `documentMapToJoinConfiguration()` component config is not considered.

please LMK, if someone has strong opinions for refactors here.